packages/trpc/server/routers/viewer/organizations/updateUser.handler.ts

```diff
@@ -94,7 +99,7 @@ export const updateUserHandler = async ({ ctx, input }: UpdateUserOptions) => {
   if (!requestedMember)
     throw new TRPCError({ code: "UNAUTHORIZED", message: "User does not belong to your organization" });
 
-  const hasUsernameUpdated = input.username !== requestedMember.user.username;
+  const hasUsernameUpdated = input.username !== requestedMember.user.profiles[0]?.username;
```

Comment:
We were simply thinking that username changed because of this always and this would cause the user's own username to be considered used by checkRegularUsername below